runtime/tests/interpreter/interpreter_test.go

```diff
@@ -10395,6 +10395,181 @@ func TestInterpretArrayFirstIndexDoesNotExist(t *testing.T) {
 	)
 }
 
+func TestInterpretArrayReverse(t *testing.T) {
+	t.Parallel()
+
+	inter := parseCheckAndInterpret(t, `
+      let xs = [1, 2, 3, 100, 200]
+      let ys = [100, 467, 297, 23]
```

Comment:
Can you add some tests for constant-sized arrays as well?

Reply:
Done, this uncovered a bug as well. The return type of function was always set to `VariableSized`. I fixed it to return whatever the type of the original array is.